src/components/ContactsModal/Body.tsx

```diff
@@ -0,0 +1,109 @@
+import React, { ChangeEvent, useEffect, useRef, useState } from 'react';
+import styles from './Body.module.scss';
+import ContactList from "../ContactList/ContactList";
+import { useAppDispatch, useAppSelector } from "../../store/hooks";
+import { getContacts } from "../../store/features/contacts/contactsSlice";
+
+interface Props {
+    country: '' | 'us',
+    onlyEven: boolean,
+}
+
+const ContactsModalBody = (props: Props) => {
+    const { country, onlyEven } = props;
+    const [query, setQuery] = useState('');
+    const [page, setPage] = useState(1);
+    const [inputTimeout, setInputTimeout] = useState<any>(null);
+    const dispatch = useAppDispatch();
+    const { status, data } = useAppSelector(state => state.contacts);
+    const { contacts, next } = data;
+    const listRef = useRef<any>(null);
+    const firstRender = useRef(true);
+    const isLoading = status === 'loading';
+
+    useEffect(() => {
+        if ((page > 1 && !next) || isLoading) return;
+
+        dispatch(getContacts({ country, query, page, onlyEven: onlyEven ? 1 : 0 }))
+    }, [page])
+
+    useEffect(() => {
+        if (firstRender.current) {
+            firstRender.current = false;
+            return;
+        }
+
+        getContactsByFilters(query, 1)
+    }, [onlyEven])
+
+    useEffect(() => {
+        if (!listRef.current) {
+            return;
+        }
+
+        listRef.current.addEventListener('scroll', scrollCallback);
+
+        return () => window.removeEventListener('scroll', scrollCallback)
+    }, [])
+
+
+    const scrollCallback = () => {
+        const scrollHeight = listRef.current.scrollHeight;
+        const scrollTop = listRef.current.scrollTop;
+        const clientHeight = listRef.current.clientHeight;
+
+        if (scrollTop === (scrollHeight - clientHeight)) {
+            setPage(prevPage => prevPage + 1);
+        }
+    }
+
+    const getContactsByFilters = (queryArg: string, pageArg: number) => {
+        if (page === 1) {
```

Comment:
cam confuzing cum faci managementul la page. Chiar era nevoie de aces if?

Reply:
iti explic in call